src/SimpleVideoCutter/MainForm.cs

```diff
@@ -615,21 +596,121 @@ private void StoreSettings()
             VideoCutterSettings.Instance.StoreSettings();
         }
 
-        private void EnqeueNewTask()
+
+        private void UnloadVideo()
+        {
+            if (vlcControl1.MediaPlayer != null)
+            {
+                vlcControl1.MediaPlayer.Stop();
+                vlcControl1.MediaPlayer.Media?.Dispose();
+                vlcControl1.MediaPlayer.Media = null;
+            }
+
+            if (videoViewHover.MediaPlayer != null)
+            {
+                videoViewHover.MediaPlayer.Stop();
+                videoViewHover.MediaPlayer.Media?.Dispose();
+                videoViewHover.MediaPlayer.Media = null;
+            }
+        }
+
+
+        private void ActionAfterTaskCompletion_ActionExecuting(object? sender, ActionExecutingEventArgs e)
+        {
+            if (e.Action is IActionRemovesOriginalFile && (e.Action as IActionRemovesOriginalFile)?.OriginalFilePath == this.fileBeingPlayed)
+            {
+                if (this.shouldNotifyIfCurrentFileIsBeingDeleted)
+                {
+                    DialogResult dialogResult = MessageBox.Show("As per the settings, the original file will be deleted after the cut was saved." +
+                        " Since you are currently playing this file, the program will skip to the next file first." +
+                        "\n\nDo you want to see this message the next time this happens?",
+                        "Playing file about to be deleted",
+                        MessageBoxButtons.YesNo
+                    );
+
+                    if (dialogResult == DialogResult.No)
+                    {
+                        this.shouldNotifyIfCurrentFileIsBeingDeleted = false;
+                    }
+                }
+
+                UnloadVideo();
+                OpenNextFileInDirectory();
+            }
+        }
+
+        private IActionAfterTaskCompletion? GetActionAfterTaskCompletion(FFmpegTask task)
+        {
+            string actionName = VideoCutterSettings.Instance.OriginalFileActionAfterCut ?? "keep";
+            IActionAfterTaskCompletion? action = null;
+
+            if (actionName == "keep" || task.InputFilePath == null)
```

Comment:
> Or maybe even better - wouldn't it be enough to assume that these actions have constructors that get InputFilePath and need to initialize on their own (access VideoCutterSettings is necessary, or show dialogs/questions to the user). In this case the only thing would be to derive concrete class name from the selected actionName (simple convention would do the job here)

I thought about that, too, and I agree. With the exception that I still would pass the task, not just its `InputFilePath` property, since I want to leave open the possibility of creating diverse actions. My actions just happen to move/delete the input file.